In [10]:
import numpy as np
import pandas as pd
import scipy.stats as stats
import matplotlib.pyplot as plt
from scipy.stats import f_oneway
from scipy.stats import levene
%matplotlib inline

In [11]:
data = pd.read_csv('data.csv')

In [12]:
# a. Checking the distribution of all the classes using one-way ANOVA
f_statistic, p_value = f_oneway(data[' Blood Pressure Before (mmHg)'], data[' Blood Pressure After (mmHg)'])
print("a. One-way ANOVA test:")
print("F-statistic:", f_statistic)
print("p-value:", p_value)

a. One-way ANOVA test:
F-statistic: 33.85555270599285
p-value: 2.3565868442707578e-08


**Intepretation**:
- The one-way ANOVA test reveals a significant difference among the means of the blood pressure before and after measurements. The high F-statistic of 33.86 and the very low p-value of 2.36e-08 indicate strong evidence against the null hypothesis of equal means across all classes.

In [14]:
# b. Checking the equality of variance using Levene's test
levene_statistic, levene_p_value = levene(data[' Blood Pressure Before (mmHg)'], data[' Blood Pressure After (mmHg)'])
print("\nb. Levene's test for equality of variance:")
print("Levene's statistic:", levene_statistic)
print("p-value:", levene_p_value)


b. Levene's test for equality of variance:
Levene's statistic: 0.18038002140150966
p-value: 0.6715080090945376


**Interpretation**:
- Levene's test for equality of variance shows that there is no significant difference in variances between the blood pressure before and after measurements. The low Levene's statistic of 0.18 and the relatively high p-value of 0.67 suggest that the variances are similar for the two groups.

In [16]:
# d. Checking the equality of means between all the classes using one-way ANOVA
f_statistic, p_value = f_oneway(data[' Blood Pressure Before (mmHg)'], data[' Blood Pressure After (mmHg)'])
print("\nd. One-way ANOVA test for equality of means between classes:")
print("F-statistic:", f_statistic)
print("p-value:", p_value)


d. One-way ANOVA test for equality of means between classes:
F-statistic: 33.85555270599285
p-value: 2.3565868442707578e-08


**Interpretation**:
- The one-way ANOVA test for equality of means indicates a significant difference in means between the blood pressure before and after measurements. The high F-statistic of 33.86 and the very low p-value of 2.36e-08 provide strong evidence against the null hypothesis of equal means across all classes.